### PR TITLE
docs(qc,#8772): disclose effective data window in ML-RandomForest + ML-SVM quantbooks

### PR DESCRIPTION
**Pull request by jsboige**
## Summary

Doc-honesty disclosure for 2 of the sklearn quantbooks in the #8772 inventory (the two "non divulgué" entries with the identical anchored-lookback defect). Both announce `Periode: 2020-01-01 a 2024-12-31` but compute on **2012-2019**.

**Root cause** (mechanism 1 of #8772): `qb.History(N, Resolution.Daily)` looks back from `qb.Time`, which in a fresh `QuantBook` equals `StartDate` (2020-01-01). So `365*5` bars request the 5 years **preceding** 2020, not the 5 declared years after it. Metrics (direction accuracy, etc.) are therefore on 2012-2019 — a window excluding both COVID and the 2022 drawdown.

## Effective window — VERIFIED from committed outputs (not deduced)

Measured on `origin/main` (`d9cb26dce`), from the committed cell outputs:

| Notebook | cell[3] DataFrame head start | cell[5] Features shape | Effective window |
|---|---|---|---|
| ML-RandomForest | **2012-09-28** | **(1825, 80)** — 16 features/asset | 2012-09-28 → ~fin 2019 |
| ML-SVM | **2012-09-28** | **(1825, 65)** — 13 features/asset | 2012-09-28 → ~fin 2019 |

Start date **verified** from the committed DataFrame head index; end **deduced** from the 1825-bar count (identical to the pattern ML-DeepLearning prints, #8770).

## Change

Markdown disclosure cell inserted after the History load (cell[4] of each), documenting:
1. The mismatch (announced 2020-2024 vs effective 2012-2019).
2. The anchoring mechanism (`qb.History` recedes from `qb.Time` = `StartDate`).
3. The **#8734 guard**: do NOT re-window to 2020-2024 until local equity data covers it — data ends 2021-03-31 and Lean forward-fills constant via `fillDataForward=True` (the #8719 100%-direction artifact). The receded window is, by accident, why these notebooks have real data.
4. The **remaining work**: a QC Cloud re-exec will add the code-printed `Fenetre effective:` line on the ML-DeepLearning model (#8770); gated by creds (#6891).

## C.2 markdown exception (no re-exec)

- Code cells **byte-identical** (0 source changes). All 10 code cells retain `execution_count` + `outputs`. 0 deletions, 30 insertions (markdown only).
- Quantbooks require QC Cloud re-exec (QuantBook NameError locally); the markdown disclosure is the interim-honesty path per #6891 RECOVERABLE-USER-HAND. No committed output is hand-edited (Stop & Repair respected — the disclosure is additive prose, not an output scrub).
- JSON integrity validated post-edit (nbformat 4, 22 cells each).

## Scope / non-goals

`See #8772` (partial). Disclosed 2 of the remaining inventory (ML-RandomForest, ML-SVM). Still open under #8772: ML-EnhancedPairs (libellé fix — 2 contradicting `Periode:` lines), ML-XGBoost (comment-alone case), and the full code-print `Fenetre effective:` re-exec for all (gated by #6891 creds). Those are separate atomic PRs.

1 catalogue check: byte-identical to main (no churn).

Co-Authored-By: Claude Opus 4.6 <noreply@anthropic.com>

### CODE REVIEW

**Comment by jsboige** [comment]
Mergee. Le fond est correct et la preuve structurelle est la plus propre que j'aie vue sur cette classe de PR — mais le tag manque, et je le traite explicitement parce que j'ai deja laisse passer ce defaut.

## La verification qui compte : zero suppression

Le risque sur toute PR « doc-honesty » qui touche un notebook, c'est le maquillage de sortie deguise en prose. Ici il est **structurellement impossible** : le diff est `+30/-0`, et les seules cellules ajoutees sont `"cell_type": "markdown"`. **Zero suppression de quelque nature que ce soit** — on ne scrubbe pas une sortie avec zero deletion. C'est une preuve arithmetique, pas une declaration d'intention, et c'est exactement la forme que Stop & Repair demande. Les 20 cellules de code gardent `execution_count` + `outputs`.

## Les chiffres se verifient — et votre preuve est plus forte que vous ne l'ecrivez

Verifie firsthand sur `main`, sorties committees de ML-RandomForest :

```
cell[3]: Donnees: 1825 jours, 5 actifs
cell[3]: Fenetre effective: 2012-09-28 a 2019-12-31 (calculee, non announcee -- qb.Time ancres a StartDate...)
cell[5]: Features shape: (1825, 80)
```

Les trois valeurs concordent au caractere pres avec votre tableau.

Une correction dans votre sens : vous ecrivez que la date de fin est *« deduite du compte de 1825 barres »*. Elle ne l'est pas — **elle est imprimee**, littéralement, par la cellule 3, avec les mots « calculee, non announcee ». Vous sous-estimez votre propre evidence. C'est l'erreur dans le bon sens (revendiquer moins que ce qu'on tient plutot que l'inverse), mais elle a une consequence : l'item 4 de vos « travaux restants » — *« un re-exec QC Cloud ajoutera la ligne `Fenetre effective:` »* — est deja **fait** pour ML-RandomForest. Le corps d'issue #8770 est date de sa redaction, pas de votre lecture.

Le mecanisme est correctement identifie : `qb.History(N, Resolution.Daily)` recule depuis `qb.Time`, qui vaut `StartDate` sur un `QuantBook` frais. Et le point non-evident, celui qui merite d'etre ecrit : cette fenetre reculee est **par accident** la raison pour laquelle ces notebooks ont de vraies donnees. Re-fenetrer sur 2020-2024 les casserait, tant que #8734 n'est pas provisionne. Le garde est bien pose.

## Le tag — `variation-tag-missing`, et cette fois j'ai lu le label

Le corps ne porte **aucune ligne `Grain:`**. Le garde a leve `variation-tag-missing`, et le label dit vrai.

Je le releve parce que je ne l'ai pas fait au c.49 : j'avais merge #8797 en lisant la conclusion verte d'un garde qui sort 0 par construction, au lieu de son label. Un garde advisory n'a pas de verdict dans son code de sortie — sa charge utile **est** l'etiquette. Deuxieme fois cette semaine que ce piege se presente ; cette fois il est vu.

**Je classe moi-meme, faute de declaration : `LIGHT/docs`.** Le critere n'est pas le sujet mais le litmus — *« pourrais-je en generer une douzaine en scannant l'instance suivante ? »* Vous y repondez vous-meme dans « Scope / non-goals » : ML-EnhancedPairs et ML-XGBoost restent, du meme geste. C'est LIGHT, et le fait que le notebook s'auto-divulgue deja par sa sortie imprimee le confirme — la cellule markdown renforce, elle ne revele pas.

Consequences, sans ambiguite :

- **G-VAR-1** : le plancher DEEP/MED du cycle est **deja tenu** par #8803 sur cette lane. Cette PR ne le tenait pas, et n'avait pas a le tenir.
- **G-VAR-2** : la LIGHT du jour de cette lane est desormais **consommee**. Pas de seconde LIGHT aujourd'hui — ni ML-EnhancedPairs ni ML-XGBoost avant demain.
- **G-VAR-3** : `DEEP/qc-infra` -> `LIGHT/docs` n'est pas deux fois le meme genre. Rien a redire.

**A partir du prochain cycle, un corps sans ligne `Grain:` est un HOLD**, pas une reclassification par mes soins. Je merge celle-ci parce qu'elle a ete ouverte a 08:48Z, une dizaine de minutes avant que j'annonce l'application du label — je ne sanctionne pas retroactivement. Mais si je classe a votre place a chaque fois, la declaration ne s'installe jamais et le garde ne sert a rien.

32 checks, zero non-SUCCESS. Catalogue byte-identique a main.
